skglm/experimental/smooth_quantile_regressor.py

```diff
@@ -0,0 +1,553 @@
+        self.initial_alpha = float(
+            initial_alpha) if initial_alpha is not None else 5.0 * self.alpha
+        self.alpha_schedule = alpha_schedule
+
+        self.solver_params = {} if solver_params is None else dict(solver_params)
+
+        # Initialize solver strategy
+        self.solver_strategy = StageBasedSolverStrategy(self.solver_params)
+
+        from skglm.experimental.quantile_huber import QuantileHuber
+        self._quantile_huber_cls = QuantileHuber
+
+        # Initialize solver - use PDCD for L1 regularization
+        if smooth_solver is None:
+            self.smooth_solver = PDCD_WS(
+                max_iter=200,
+                tol=1e-8,
+                fit_intercept=False,
+                warm_start=True,
+                p0=200
+            )
+        else:
+            self.smooth_solver = smooth_solver
+
+        # Ensure consistent solver settings
+        if not hasattr(self.smooth_solver, 'warm_start'):
+            self.smooth_solver.warm_start = False
+
+    def _initialize_pdcd(self, X, y):
+        """Initialize PDCD solver with good primal and dual variables."""
+        _, n_features = X.shape
+
+        if hasattr(self, 'coef_') and self.coef_ is not None:
+            w = self.coef_.copy()
+        else:
+            w = np.zeros(n_features)
+
+        residuals = y - X.dot(w)
+        dual = np.where(
+            residuals > 0,
+            self.quantile,
+            self.quantile - 1
+        )
+
+        return w, dual
+
+    def _get_solver_for_stage(self, delta, stage, n_features):
+        """Get solver with parameters adapted to current stage.
+
+        Uses the solver strategy to create and configure a solver appropriate
+        for the current smoothing stage.
+
+        Parameters
+        ----------
+        delta : float
+            Current smoothing parameter.
+        stage : int
+            Current stage number (0-indexed).
+        n_features : int
+            Number of features in the dataset.
+
+        Returns
+        -------
+        solver : BaseSolver
+            Configured solver instance for the current stage.
+        """
+        # Get base solver configuration
+        solver = self.solver_strategy.create_solver_for_stage(
+            self.smooth_solver, delta, stage, n_features)
+
+        # Additional quantile-specific configuration
+        if hasattr(solver, 'tol'):
+            # Scale tolerance based on quantile error
+            solver.tol = min(solver.tol, self.quantile_error_threshold * 0.1)
+
+        return solver
+
+    def fit(self, X, y):
+        """Fit the model according to the given training data."""
+        # Basic validation
+        n_samples, n_features = X.shape
+        if len(y) != n_samples:
+            raise ValueError(f"X has {n_samples} samples, but y has {len(y)} samples")
+        is_sparse = sparse.issparse(X)
+
+        # Center data to handle intercept manually
+        y_mean = np.mean(y)
+        if is_sparse:
```

Comment:
this is to be avoided at all cost ; all our solvers support sparse X ! also, no need to center